Decimal numbers, such as 1.67, are represented by a floating-point type. A difference from integer numbers is that a decimal number has a limit of 15 digits of precision (e.g. the number of digits after the decimal). This is because, with less precision, the calculations run faster. You just need to be aware of this in case your data has very high precision, Python will round your number if you have more than 15 digits after the decimal.

In [1]:
#a decimal number assigned to d variable
d=9.5889
d

9.5889

In [2]:
#type
type(d)

float

In [3]:
#Precision limit(15 degits)
10/3

3.3333333333333335

10/3 is equal to 3.33333... until infinity, but Python has rounded the value. Also, note that the last digit has rounded to 5 rather than 4. This occurs when trying to represent very high precision numbers and is called a roundoff error. As you work with your data, you need to be aware of the possible impact of roundoff errors.

In [4]:
#Python rounds after 15 digits
7.3333333333333335

7.333333333333334

In [8]:
a = 2.5788888888888888888
b = 5.678
c = a*b
b = 16.89
print(a,b,c)

2.578888888888889 16.89 14.64293111111111
